index.bs

```diff
@@ -1490,16 +1490,21 @@ a numbered step. If outdented, it (today) is rendered as a bullet in the midst o
 1. Let |credTypesAndPubKeyAlgs| be a new [=list=] whose [=list/items=] are pairs of {{PublicKeyCredentialType}} and
     a {{COSEAlgorithmIdentifier}}.
 
-1. [=list/For each=] |current| of <code>|options|.{{PublicKeyCredentialCreationOptions/pubKeyCredParams}}</code>:
+1. If <code>|options|.{{PublicKeyCredentialCreationOptions/pubKeyCredParams}}</code> [=list/is empty=]:
+
+    1. [=list/Append=] the pair of {{PublicKeyCredentialType}} with value {{public-key}} and a
+        {{COSEAlgorithmIdentifier}} with value <code>-7</code> ("ES256") to |credTypesAndPubKeyAlgs|.
+
+1. Otherwise, [=list/for each=] |current| of <code>|options|.{{PublicKeyCredentialCreationOptions/pubKeyCredParams}}</code>:
 
     1. If <code>|current|.{{PublicKeyCredentialParameters/type}}</code> does not contain a {{PublicKeyCredentialType}} supported
         by this implementation, then [=continue=].
     1. Let |alg| be <code>|current|.{{PublicKeyCredentialParameters/alg}}</code>.
     1. [=list/Append=] the pair of <code>|current|.{{PublicKeyCredentialParameters/type}}</code> and |alg| to
         |credTypesAndPubKeyAlgs|.
 
-1. If |credTypesAndPubKeyAlgs| [=list/is empty=] and <code>|options|.{{PublicKeyCredentialCreationOptions/pubKeyCredParams}}</code>
-    [=list/is not empty=], return a {{DOMException}} whose name is "{{NotSupportedError}}", and terminate this algorithm.
+1. If |credTypesAndPubKeyAlgs| [=list/is empty=], return a {{DOMException}} whose name is
+    "{{NotSupportedError}}", and terminate this algorithm.
```

Comment:
Is this step even needed any longer?  ISTM that `credTypesAndPubKeyAlgs` will now never be empty ...?

Reply:
It's necessary. It can still be empty if `pubKeyCredParams` is not empty and none of its algorithms are supported.

Reply:
What I'm thinking now is that perhaps it's better to move it to the "is non-zero" branch of the switch.

Reply:
Done.

Reply:
> It's necessary. It can still be empty if `pubKeyCredParams` is not empty and none of its algorithms are supported.

Right. doh.

Reply:
fyi/fwiw, I did not obviously see support for `[=list/is not empty=]` in infra.spec.whatwg.org (but should have looked at index.bs more closely because that construct is in there (doh)), hence suggesting leveraging the list's `size` for the switch.  Either works so this is fine but would not object to using `is (not) empty` instead.  no big deal either way.